core-common/src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -230,7 +230,7 @@ public void repartition(final String topic, final int partitionsNumber) throws C
             if (!Boolean.TRUE.equals(areNewPartitionsAdded)) {
                 throw new TopicConfigException(String.format("Failed to repartition topic to %s", partitionsNumber));
             }
-            final Producer<byte[], byte[]> producer = kafkaFactory.takeProducer();
+            final Producer<byte[], byte[]> producer = kafkaFactory.takeProducer(topic);
             kafkaFactory.terminateProducer(producer);
             kafkaFactory.releaseProducer(producer);
```

Comment:
Why don't we throw this piece of code away?

Reply:
you are welcome to open another PR for that